frontend/src/MainPage/OnLoadUIFolder/TextArea.tsx

```diff
@@ -1,36 +1,23 @@
 import { InputBoxArrowLogo } from "../svg/InputBoxArrowLogo";
-import { useState,useContext, useEffect } from "react";
+import { useContext } from "react";
 import { MainPageContext } from "../context/MainPageContext";
-
-
+import useTextAnimation from "../../shared/Hooks/useTextAnimation";
 
 export const TextArea = () => {
-  const [text,setText] = useState("");
 
   const message = "Where can I find a young, hard working, and passionate developer?";
   const mainPageContext = useContext(MainPageContext);
   const {setIsOnLoadUIFinished} = mainPageContext;
 
-  useEffect(() => {
-    const timer = setInterval(() => {
-      setText((prevText) => {
-        // If all characters have been added, stop the interval
-        if (prevText.length === message.length) {
-          clearInterval(timer);
-          // We need to do this in Order to prevent violating the React Hooks rule of having side effects
-          setTimeout(() => setIsOnLoadUIFinished(true), 0)
-          return prevText;
-        }
-        // Add the next character from the message
-        return prevText + message.charAt(prevText.length);
-      });
-    }, 40); // adjust the speed of typing here
-    return () => clearInterval(timer);
-  }, []);
+  const {animatedText} = useTextAnimation({
+    message:message,
```

Comment:
nit: i think you can do 
```
{
    message,
    FunctionToTriggerUseContext: () => setIsOnLoadUiFinished(true)
}
```
and this would be the same thing as `message:message`